returnn/tf/updater.py

```diff
@@ -17,9 +17,11 @@
 from returnn.tf.util.basic import tf_version_tuple, assert_min_tf_version, CustomUpdate, add_check_numerics_ops
 
 Optimizer = tf_compat.v1.train.Optimizer
+if tf_compat.v2:
+  KerasOptimizer = tf_compat.v2.optimizers.Optimizer
```

Comment:
```suggestion
KerasOptimizer = tf_compat.v2.optimizers.Optimizer if if tf_compat.v2 else None
```

Reply:
Btw, with this, also later the code `allowed_types += (tf_compat.v2.keras.optimizers.Optimizer,)` should be changed to `allowed_types += (KerasOptimizer,)

Reply:
Actually, is `tf_compat.v2.keras.optimizers.Optimizer` the same as `tf_compat.v2.optimizers.Optimizer`? If not, I think we should reflect this more clearly in the name.

Reply:
Basically all further use of `tf_compat.v2.keras.optimizers.Optimizer` (or `tf_compat.v2.optimizers.Optimizer`) should be replaced by `KerasOptimizer`, and `if tf_compat.v2` should be replaced by `if KerasOptimizer`.